Fix issue 13 - Add JobId parameterset to Get-xDscConfiguration

dantraMSFT opened the pull request:
Fix https://github.com/PowerShell/xDscDiagnostics/issues/13
- add JobId parameter set
- Add unit tests for success, invalid guid, and job not found

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xdscdiagnostics/38)
<!-- Reviewable:end -->

msftclas commented:
Hi __@dantraMSFT__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Dan Travison). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

narrieta commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[Tests/Unit/CollectDscDiagnotics.tests.ps1, line 296 at r1](https://reviewable.io:443/reviews/powershell/xdscdiagnostics/38#-KX8SudjKAk4Qa41sBN2:-KX8SudjKAk4Qa41sBN3:b-cklsrj) ([raw file](https://github.com/powershell/xdscdiagnostics/blob/bc37a3e24065ca120127f84833a38ed4e701f6a8/Tests/Unit/CollectDscDiagnotics.tests.ps1#L296)):*
> Get-ChildItem

This is a very common function; try to restrict when your mock is used, maybe with filters

---

*[Tests/Unit/CollectDscDiagnotics.tests.ps1, line 316 at r1](https://reviewable.io:443/reviews/powershell/xdscdiagnostics/38#-KX8TGKIDL2610S8pJ2H:-KX8TGKIDL2610S8pJ2I:bjzcz01) ([raw file](https://github.com/powershell/xdscdiagnostics/blob/bc37a3e24065ca120127f84833a38ed4e701f6a8/Tests/Unit/CollectDscDiagnotics.tests.ps1#L316)):*
> -verbose

is that verbose intentional? the other test doesn't have it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdscdiagnostics/38)*
<!-- Sent from Reviewable.io -->

narrieta commented:
Minor comments

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xdscdiagnostics/38#-:-KX8TVFHurGV_1gaBiA5:bgevdps)*
<!-- Sent from Reviewable.io -->

dantraMSFT commented:
No, removed -verbose and added a -Path parameter filter to Get-ChildItem

---
In reply to: [262113089](https://github.com/PowerShell/xDscDiagnostics/pull/38#issuecomment-262113089) [](ancestors = 262113089)
